Change Tide's 'Missing' column to indicate that tests may be failing.

cjwagner opened the pull request:
If a user sees that their PR is listed as `Missing` on the Tide dashboard they probably won't know what that means or may think it means that their PR could not be found by Tide.

`Missing` really means one of three things:
- Tide could not find test results or running tests for at least one job for the PR (PR has not run all test jobs)
- Tide found tests associated with the PR for all jobs, but some job does not have results or is not currently running against the current repo HEAD. (PR does not have up to date run for each test job)
- Tide found up-to-date tests for the PR, tested against the current repo HEAD, but at least one such test failed. (PR has up-to-date test that is failing)

The column title should indicate that PRs in the column are either failing tests or have old test results that are no longer valid.

cjwagner commented:
/cc @spxtr @BenTheElder

stevekuznetsov commented on prow/cmd/deck/static/tide.html:
```diff
@@ -34,7 +34,7 @@ <h4>Your Pull Request must also match one of these GitHub search queries (you ca
                 <th>Passing</th>
                 <th>Pending</th>
-                <th>Missing</th>
+                <th>Failing/Untested against HEAD</th>
```
`HEAD` is technically a clone-local reference and even the more general `HEAD` meaning `master` isn't correct when the PR targets a non-`master` branch so maybe just `"Failing or Untested"` as a header?

cjwagner replied:
Ok, that makes sense.
I'm thinking that the case in which a PR has all green status checks against an older commit could still be confusing since the PR will appear to be both tested and passing.  Thats probably too much information to convey in a column header though. It probably makes more sense to include that at the top of the Tide dashboard. Maybe as part an addition to:
> All GitHub statuses on a Pull Request must be passing / green for automatic merge ✓

cjwagner commented:
I changed the column header as requested and added some more explanation to the top of the page. Its a little verbose, but it seems like important information to include.
![screenshot from 2018-01-04 16 02 27](https://user-images.githubusercontent.com/5334145/34589987-d6328f56-f168-11e7-8910-a41eb978d4c0.png)

kargakis commented:
Can we make the info section expandable in a follow-up? It has ended up being a wall of text and github labels. Most people landing on this page won't bother reading through and we make it harder to show them what they really want to see (what's going on with their PR).

BenTheElder commented:
> Can we make the info section expandable in a follow-up? It has ended up being a wall of text and github labels. Most people landing on this page won't bother reading through and we make it harder to show them what they really want to see (what's going on with their PR).

Can we take a look at this now? This should be easy to add.

cjwagner commented:
I made the information section collapsible. Here is what the page initially looks like now:
![screenshot from 2018-01-08 15 29 48](https://user-images.githubusercontent.com/5334145/34698129-635a08bc-f48b-11e7-8fc1-6a0e2a2d936b.png)

And here is what it looks like after being expanded:
![screenshot from 2018-01-08 15 30 12](https://user-images.githubusercontent.com/5334145/34698144-7a7bba90-f48b-11e7-9dfd-bc56e88ce318.png)

Thoughts?

stevekuznetsov commented:
Why not use 

```
<details><summary>Merge Requirements:</summary>....</details>
```
?

I think most browsers will render that nicely with a little expand button and everything

BenTheElder commented:
FWIW https://caniuse.com/#feat=details says 88.57% globally, but not IE / Edge (any version) and Opera Mini. 

Edge is supposedly in development but this page already requires JavaScript to function so I'm not sure we gain much from `<details>` over toggling `display: none` and the expansion icon ourselves (or something along those lines...).

@qhuynh96 want to take a stab at making this nicer? :-)

qhuynh96 commented:
@cjwagner  @stevekuznetsov  @BenTheElder  I would suggest to have a "i" (information) button which on clicked will bring up modal that contains the information. The button would float right on the header, perhaps, so that the body will contain a list of repos only.

stevekuznetsov commented:
@BenTheElder good catch... I'm pretty far away from even thinking about IE and Edge, I'll admit :)

The content LGTM here.

BenTheElder commented:
> @BenTheElder good catch... I'm pretty far away from even thinking about IE and Edge, I'll admit :)

I'm admittedly not *particularly* worried about supporting them either, but if it already works for them I feel like we need a pretty good reason to break them. :man_shrugging:

fejta commented:
I think we already have a bunch of tooling that uses `<details>` like prow, triage dashboard, etc

BenTheElder commented:
@cjwagner Looks like this needs a rebase. I'd also consider setting `pointer` on the clickable area for desktop users.

cjwagner commented:
Rebased and set `cursor` to `pointer` for the clickable area. PTAL

![screenshot from 2018-01-22 16 38 41](https://user-images.githubusercontent.com/5334145/35252150-e703e4b8-ff93-11e7-9d73-f7ca2b06db6f.png)
![screenshot from 2018-01-22 16 38 58](https://user-images.githubusercontent.com/5334145/35252152-e9829982-ff93-11e7-8080-dbbbbaf00eb5.png)

stevekuznetsov commented:
/hold cancel
